contracts/Removal.sol

```diff
@@ -33,18 +34,27 @@ contract Removal is
     bool list;
   }
 
-  uint256 private _tokenIdCounter;
+  /**
+   * @notice The EscrowedNORI contract that manages escrowed tokens.
+   */
+  EscrowedNORI private _eNORI;
+  uint256 public tokenIdCounter;
   string public name; // todo why did I add this
   mapping(uint256 => uint256) public indexToTokenId; // todo consider how we're keeping track of the number and order of ids, ability to iterate
   mapping(uint256 => bool) private _tokenIdExists;
+  mapping(uint256 => uint256) private _idToEscrowScheduleStartTime;
 
-  function initialize() public virtual initializer {
+  function initialize() external virtual initializer {
     super.initialize("https://nori.com/api/removal/{id}.json");
     __ERC1155Supply_init_unchained();
-    _tokenIdCounter = 0;
+    tokenIdCounter = 0;
     name = "Removal";
   }
 
+  function initializeEscrowedNORI(address escrowedNORIAddress) external {
+    _eNORI = EscrowedNORI(escrowedNORIAddress);
```

Comment:
that's because there's a circular initialization between `Removal.sol` and `RestrictedNORI.sol` ... so one of them has to come first and that contract won't yet have the address of the other in the initializer.  This all works completely differently now as the market contract serves as a registry for all satellite addresses, and then after its deployment the deployment script calls something on the market contract that sets up all the connections.  This call can't happen during the market contract initialization because we have to first deploy the market contract and get its address before we can use its address to call into `Removal.sol` and `RestrictedNORI.sol` to add the market contract as a `CONTRACT_INITIALIZER_ROLE` basically giving it permission to do that setup.  Maybe there's a way to do this so that the setup performed by the market contract can happen during its actual initialization, which I think would require knowing the address of the market contract before it's deployed, so that it can be granted roles with the other contracts before it's deployed.